src/scripting/lua_ui.h

```diff
@@ -88,28 +98,52 @@ class LuaPanel : public LuaUiModuleClass {
 	int indicate(lua_State* L);
 #endif
 	int create_child(lua_State* L);
+	int get_child(lua_State* L);
 
 	/*
 	 * C Methods
 	 */
 	static UI::Panel* do_create_child(lua_State* L, UI::Panel* parent, UI::Box* as_box);
 };
 
-class LuaButton : public LuaPanel {
+class LuaNamedPanel : public LuaPanel {
+public:
+	LUNA_CLASS_HEAD(LuaNamedPanel);
+
+	LuaNamedPanel() = default;
+	explicit LuaNamedPanel(UI::Panel* p) : LuaPanel(p) {
+	}
+	explicit LuaNamedPanel(lua_State* L) : LuaPanel(L) {
+	}
+	~LuaNamedPanel() override = default;
+
+	/*
+	 * Properties
+	 */
+	int get_name(lua_State* L);
+
+	/*
+	 * C Methods
+	 */
+	UI::NamedPanel* get() {
+		return dynamic_cast<UI::NamedPanel*>(panel_);
+	}
+};
+
+class LuaButton : public LuaNamedPanel {
```

Comment:
why ares some of these extensions of named panel and some are not?
which panels (ui elements can be named and which can not be named?

Reply:
Currently the only components derived from NamedPanel are `Tab Window BaseDropdown Button` and their own derivatives. The Lua APIs for those all therefore inherit from `LuaNamedPanel`. All other types of components are currently derived directly from `Panel`, though I would like to (in a separate big branch) remove NamedPanel and make all panels named.

Reply:
Currently the only components derived from NamedPanel are `Tab Window BaseDropdown Button` and their own derivatives. The Lua APIs for those all therefore inherit from `LuaNamedPanel`. All other types of components are currently derived directly from `Panel`, though I would like to (in a separate big branch) remove NamedPanel and make all panels named.